app/forms/idv/inherited_proofing/va/form.rb

```diff
@@ -30,6 +42,30 @@ def user_pii
           user_pii[:zipcode] = address_zip
           user_pii
         end
+
+        def service_error?
+          service_error.present?
+        end
+
+        private
+
+        def extra
+          # Make sure the service-related error gets passed along
+          # to our FormResponse upon #submit.
+          { service_error: service_error } if service_error?
+        end
+
+        def add_service_error_if
```

Comment:
Not sure why you're using the term "hijack" to describe overriding a method in a super class, which is simply a feature of OOP. 

I can certainly change the method name if you feel that strong about it. However, should we be necessarily binding our method names to that level of detail in the logic? If the criteria to add a service error message changes from simply `#present?`, would it therefore follow that I should change the method name to conform to the new logic? The method is pretty simple; it's really only 2 lines. 

No problem either way. You previous comment, if I removed the `#extra` method in the base class and simply override the `#submit` method, I can call it from there. But, to me, I think this may be more obvious what's going on:

```
before_validation :add_service_error, if: service_error?
validates(*required_fields, presence: true, unless: :service_error?)
```

Reply:
Tentatively addressed.